test/integration/testConnection.js

```diff
@@ -10,1044 +10,1179 @@ const Util = require('./../../lib/util');
+    });
 
-// Skipped - requires manual interaction to enter credentials on browser
-describe.skip('Connection test - external browser', function () {
-  this.timeout(10000);
+    it('Wrong Username', function (done)
+    {
+      var connection = snowflake.createConnection(connOption.wrongUserName);
+      connection.connect(function (err)
+      {
+        assert.ok(err, 'Username is an empty string');
+        assert.equal('Incorrect username or password was specified.', err["message"]);
+        done();
+      });
+    });
 
-  it('Simple Connect', function (done) {
-    const connection = snowflake.createConnection(connOption.externalBrowser);
+    it('Wrong Password', function (done)
+    {
+      var connection = snowflake.createConnection(connOption.wrongPwd);
+      connection.connect(function (err)
+      {
+        assert.ok(err, 'Password is an empty string');
+        assert.equal('Incorrect username or password was specified.', err["message"]);
+        done();
+      });
+    });
 
-    connection.connectAsync(function (err, connection) {
-      try {
-        assert.ok(connection.isUp(), 'not active');
-        testUtil.destroyConnection(connection, function (err, r) {
-          try {
-            assert.ok(!connection.isUp(), 'not active');
-            done();
-          } catch (err) {
-            done(err);
-          }
+    it('Multiple Client', function (done)
+    {
+      const totalConnections = 10;
+      const connections = [];
+      for (let i = 0; i < totalConnections; i++)
+      {
+        connections.push(snowflake.createConnection(connOption.valid));
+      }
+      let completedConnection = 0;
+      for (let i = 0; i < totalConnections; i++)
+      {
+        connections[i].connect(function (err, conn)
+        {
+          testUtil.checkError(err);
+          conn.execute({
+            sqlText: 'select 1',
+            complete: function (err)
+            {
+              testUtil.checkError(err);
+              testUtil.destroyConnection(conn, function ()
+              {
+              });
+              completedConnection++;
+            }
+          });
         });
-      } catch (err) {
-        done(err);
       }
+      const sleepMs = 500;
+      const maxSleep = 60000;
+      let sleepFromStartChecking = 0;
+
+      const timeout = () => setTimeout(() =>
+      {
+        sleepFromStartChecking += sleepMs;
+        if (completedConnection === totalConnections)
+        {
+          done();
+        }
+        else if (sleepFromStartChecking <= maxSleep)
+        {
+          timeout();
+        }
+        else
+        {
+          done(`Max after ${maxSleep} it's expected to complete ${totalConnections} but completed ${completedConnection}`)
+        }
+      }, sleepMs);
+
+      timeout();
     });
   });
 
+  // Skipped - requires manual interaction to enter credentials on browser
+  describe.skip('Connection test - external browser', function () {
```

Comment:
those tests are moved to if branch - do we need them here with skip? the same also for other cases where the tests are duplicated but skipped

maybe for cleaner implementation let's create new file e.g. `testManualConnection.js` and move all those tests (with env variable checking if) to new file and in `testConnection` let's leave only automatic tests